### PR TITLE
bugfix: do not proxy connections without an xid

### LINKED ISSUE #3072: 使用XA模式时，没有启动全局事务也会自动进行代理
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
使用XA模式时，如果程序设置了类似健康检查，或者是设置了spring.datasource.durid.validation-query: SELECT 1 FROM DUAL，也会自动进行代理

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:
```
Just paste your stack trace here!
```
运行时的配置：
image
image
mysql8.0.11

image
image

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

### Ⅵ. Environment:

- JDK version : 1.8
- OS : win7
- Others:

### PR DESCRIPTION
**Pull request by a364176773**
### Ⅱ. Does this pull request fix one issue?

fixes #3072

### CODE REVIEW

File: rm-datasource/src/main/java/io/seata/rm/datasource/xa/DataSourceProxyXA.java [hunk 1]
```diff
@@ -56,6 +56,20 @@ public Connection getConnection(String username, String password) throws SQLExce
     }
 
     protected Connection getConnectionProxy(Connection connection) throws SQLException {
+        if (!RootContext.inGlobalTransaction()) {
+            return connection;
+        }
+        return getConnectionProxyXA(connection);
+
```
Comment (wangliang181230) [inline rm-datasource/src/main/java/io/seata/rm/datasource/xa/DataSourceProxyXA.java:63]: 多余的空行

**Review by wangliang181230 — APPROVED** [review]
LGTM

File: rm-datasource/src/main/java/io/seata/rm/datasource/xa/DataSourceProxyXA.java (hunk 1 above)
Comment (a364176773) [reply rm-datasource/src/main/java/io/seata/rm/datasource/xa/DataSourceProxyXA.java:63]: ok

**Review by l81893521 — APPROVED** [review]
LGTM for @wangliang181230

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**